sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -955,8 +938,19 @@ static int _create_candidate_set(const gnrc_netif_t *netif,
               ipv6_addr_to_str(addr_str, tmp, sizeof(addr_str)));
         /* "In any case, multicast addresses and the unspecified address MUST NOT
          *  be included in a candidate set."
+         *
+         * flags are set if not unspecfied and multicast addresses are in
+         * `netif->ipv6.groups` so not considered here.
          */
         if ((netif->ipv6.addrs_flags[i] == 0) ||
+            /* https://tools.ietf.org/html/rfc4862#section-2:
+             *  A tentative address is not considered assigned to an interface
+             *  in the usual sense.  An interface discards received packets
+             *  addressed to a tentative address, but accepts Neighbor Discovery
+             *  packets related to Duplicate Address Detection for the tentative
+             *  address.
+             *  (so don't consider tentative addresses for source address
+             *  selection) */
             gnrc_netif_ipv6_addr_dad_trans(netif, i)) {
```

Comment:
I know this line is not part of this PR, but could you clarify why addresses are not considered that have `gnrc_netif_ipv6_addr_dad_trans() != 0`? According to the doc of this function, the return value is:
```
the number of duplicate address detection transmissions already performed
```
How can we deduce that the address is tentative from that result?

Reply:
The number of DAD transmissions is encoded in the tentative state in our DAD implementation (that's why you see `TNT[1..3]` with the `ifconfig` command, not just `TNT`). So a value of 0 for already performed DAD transmissions means the address is *not* tentative. As the NS for DAD is sent as soon as the address is added, this is also true if you don't have the knowledge of such GNRC internals ;)

Reply:
See also doc for tentative state:

https://github.com/RIOT-OS/RIOT/blob/55adbee48808bda975dd5ee420c605bbddf4c94c/sys/include/net/gnrc/netif/ipv6.h#L48-L55